server/queues/processors/WebhookProcessor.test.ts

```diff
@@ -0,0 +1,166 @@
+import fetchMock from "jest-fetch-mock";
+import { v4 as uuidv4 } from "uuid";
+import { WebhookDelivery } from "@server/models";
+import {
+  buildUser,
+  buildWebhookDelivery,
+  buildWebhookSubscription,
+} from "@server/test/factories";
+import { flushdb } from "@server/test/support";
+import WebhookProcessor from "./WebhookProcessor";
+
+beforeEach(() => flushdb());
+beforeEach(() => {
+  jest.resetAllMocks();
+  fetchMock.resetMocks();
+});
+
+const ip = "127.0.0.1";
+
+fetchMock.enableMocks();
+
+describe("WebhookProcessor", () => {
+  test("should hit the subscription url and record a delivery", async () => {
+    const subscription = await buildWebhookSubscription({
+      url: "http://example.com",
+      events: ["*"],
+    });
+    const signedInUser = await buildUser({ teamId: subscription.teamId });
+    const processor = new WebhookProcessor();
+    await processor.perform({
+      name: "users.signin",
+      userId: signedInUser.id,
+      teamId: subscription.teamId,
+      actorId: signedInUser.id,
+      ip,
+    });
+
+    expect(fetchMock).toHaveBeenCalledTimes(1);
+    expect(fetchMock).toHaveBeenCalledWith(
+      "http://example.com",
+      expect.anything()
+    );
+    const parsedBody = JSON.parse(
+      fetchMock.mock.calls[0]![1]!.body!.toString()
+    );
+    expect(parsedBody.webhookSubscriptionId).toBe(subscription.id);
+    expect(parsedBody.event).toBe("users.signin");
+    expect(parsedBody.teamId).toBe(subscription.teamId);
+    expect(parsedBody.payload.id).toBe(signedInUser.id);
+    expect(parsedBody.payload.model).toBeDefined();
```

Comment:
This was the best way I could figure out the grab the JSON out of the fetch mock

Don't love all the non-null assertions so if anyone knows a way to clean it up let me know!